Extensions/3D/CustomRuntimeObject3D.ts

```diff
@@ -0,0 +1,377 @@
+    getAABBMinZ(): number {
+      return this.getDrawableZ();
+    }
+
+    /**
+     * Return the top Z of the object.
+     * Rotations around X and Y are not taken into account.
+     */
+    getAABBMaxZ(): number {
+      return this.getDrawableZ() + this.getDepth();
+    }
+
+    /**
+     * Set the object rotation on the X axis.
+     *
+     * This is an Euler angle. Objects use the `ZYX` order.
+     */
+    setRotationX(angle: float): void {
+      this._rotationX = angle;
+      this.getRenderer().updateRotation();
+    }
+
+    /**
+     * Set the object rotation on the Y axis.
+     *
+     * This is an Euler angle. Objects use the `ZYX` order.
+     */
+    setRotationY(angle: float): void {
+      this._rotationY = angle;
+      this.getRenderer().updateRotation();
+    }
+
+    /**
+     * Get the object rotation on the X axis.
+     *
+     * This is an Euler angle. Objects use the `ZYX` order.
+     */
+    getRotationX(): float {
+      return this._rotationX;
+    }
+
+    /**
+     * Get the object rotation on the Y axis.
+     *
+     * This is an Euler angle. Objects use the `ZYX` order.
+     */
+    getRotationY(): float {
+      return this._rotationY;
+    }
+
+    /**
+     * Turn the object around the scene x axis at its center.
+     * @param deltaAngle the rotation angle
+     */
+    turnAroundX(deltaAngle: float): void {
+      const axisX = gdjs.CustomRuntimeObject3D._temporaryVector;
+      axisX.set(1, 0, 0);
+
+      const mesh = this.get3DRendererObject();
+      mesh.rotateOnWorldAxis(axisX, gdjs.toRad(deltaAngle));
+      this._rotationX = gdjs.toDegrees(mesh.rotation.x);
+      this._rotationY = gdjs.toDegrees(mesh.rotation.y);
+      this.setAngle(gdjs.toDegrees(mesh.rotation.z));
+    }
+
+    /**
+     * Turn the object around the scene y axis at its center.
+     * @param deltaAngle the rotation angle
+     */
+    turnAroundY(deltaAngle: float): void {
+      const axisY = gdjs.CustomRuntimeObject3D._temporaryVector;
+      axisY.set(0, 1, 0);
+
+      const mesh = this.get3DRendererObject();
+      mesh.rotateOnWorldAxis(axisY, gdjs.toRad(deltaAngle));
+      this._rotationX = gdjs.toDegrees(mesh.rotation.x);
+      this._rotationY = gdjs.toDegrees(mesh.rotation.y);
+      this.setAngle(gdjs.toDegrees(mesh.rotation.z));
+    }
+
+    /**
+     * Turn the object around the scene z axis at its center.
+     * @param deltaAngle the rotation angle
+     */
+    turnAroundZ(deltaAngle: float): void {
+      const axisZ = gdjs.CustomRuntimeObject3D._temporaryVector;
+      axisZ.set(0, 0, 1);
+
+      const mesh = this.get3DRendererObject();
+      mesh.rotateOnWorldAxis(axisZ, gdjs.toRad(deltaAngle));
+      this._rotationX = gdjs.toDegrees(mesh.rotation.x);
+      this._rotationY = gdjs.toDegrees(mesh.rotation.y);
+      this.setAngle(gdjs.toDegrees(mesh.rotation.z));
+    }
+
+    /**
+     * @return the internal width of the object according to its children.
+     */
+    getUnscaledDepth(): float {
```

Comment:
Is it worth making this `private _getUnscaledDepth`?

Reply:
It would only save a substraction and `_maxZ` and `_minZ` are still necessary for `getDrawableZ` and `getUnscaledCenterZ`